src/main/java/core/basesyntax/service/OperationsStrategyImpl.java

```
@@ -0,0 +1,22 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exception.FruitShopException;
+import core.basesyntax.model.Operation;
+import java.util.Map;
+
+public class OperationsStrategyImpl implements OperationsStrategy {
+    private final Map<Operation, OperationHandler> handlers;
+
+    public OperationsStrategyImpl(Map<Operation,OperationHandler> handlers) {
+        this.handlers = handlers;
```

Constructor does not validate the input map or make a defensive/unmodifiable copy. The assignment requires services to validate non-null inputs and to follow encapsulation principles. Add a null-check for `handlers` and store an unmodifiable defensive copy (for example `this.handlers = Collections.unmodifiableMap(new EnumMap<>(handlers));` or `Map.copyOf(handlers)`). This prevents external mutation and fails fast on invalid input. Current constructor: `public OperationsStrategyImpl(Map<Operation,OperationHandler> handlers) { this.handlers = handlers; }`.